src/elements/progress-spinner/progress-spinner.md

```diff
@@ -0,0 +1,51 @@
+### Indeterminate
+
+Indicates infinite loading.
+
+```jsx
+<ProgressSpinner />
+```
+
+### Controlled
+
+Indicates the actual progress value.
+
+```jsx
+import { useEffect, useRef, useState } from 'react'
+import Button from '../button';
+
+const AnimatedProgress = () => {
+  const timeoutRef = useRef()
+  
+  const [value, setValue] = useState(0)
+  const maxValue = 100
+  
+  useEffect(() => {
+    if (value < maxValue) {
+      timeoutRef.current = setTimeout(() => {
+        setValue(value + 1)
+      }, 20)
+    }
+    
+    return () => clearTimeout(timeoutRef.current)
+  })
```

Comment:
```suggestion
  }, [value])
```

Reply:
I am not sure this is what I want. Anyway, the component does not render between changing values and the effect triggers value change anyway.

How do you see this? Why adding the value dep would be more correct?

Reply:
The same as [here](https://github.com/oacore/design/pull/204#discussion_r444811850). It's just correctness of hook usage nothing else. If you extend your example (adding new state) in the future it won't work as expected. :)